### ISSUE TITLE
Where can I find your training code?

### ISSUE THREAD

**Issue by johnnylu305** [sunwj/CAR#10]
Hi, nice to meet you. I am reading your paper and in your paper you mentioned that you will release the code here. However, I cannot find the training code. Where can I find your training code?

**Comment by yuqian2019** [comment]
Do u got the train code？

**Comment by johnnylu305** [comment]
@yuqian2019 No, I do not. I tried to write it by myself. But, the performance is not good. Perhaps, something wrong in my code. I hope the authors can share it, but I got no response from them.

**Comment by johnnylu305** [comment]
@yuqian2019 You can find my personal training code in my repo. But, the performance is not good. Again, probably something wrong in my code.

**Comment by yuqian2019** [comment]
> @yuqian2019 You can find my personal training code in my repo. But, the performance is not good. Again, probably something wrong in my code.

ok, thank u very much. I will view ur code, and if i find any error, I will mention u.

**Comment by vardenppp** [comment]
> @yuqian2019 No, I do not. I tried to write it by myself. But, the performance is not good. Perhaps, something wrong in my code. I hope the authors can share it, but I got no response from them.

hello，i tried to use your training code, but i met some mistakes.    
"img, kernels, offsets_h, offsets_v, = ctx.saved_tensors
ValueError: not enough values to unpack (expected 4, got 0)"
So how to solve it.
thank you!

**Comment by vardenppp** [comment]
> > @yuqian2019 No, I do not. I tried to write it by myself. But, the performance is not good. Perhaps, something wrong in my code. I hope the authors can share it, but I got no response from them.
> 
> hello，i tried to use your training code, but i met some mistakes.
> "img, kernels, offsets_h, offsets_v, = ctx.saved_tensors
> ValueError: not enough values to unpack (expected 4, got 0)"
> So how to solve it.
> thank you!

Oh!, i found it, i have solved it, thank u.

**Comment by yuqian2019** [comment]
> > > @yuqian2019 No, I do not. I tried to write it by myself. But, the performance is not good. Perhaps, something wrong in my code. I hope the authors can share it, but I got no response from them.
> > 
> > 
> > hello，i tried to use your training code, but i met some mistakes.
> > "img, kernels, offsets_h, offsets_v, = ctx.saved_tensors
> > ValueError: not enough values to unpack (expected 4, got 0)"
> > So how to solve it.
> > thank you!
> 
> Oh!, i found it, i have solved it, thank u.

ok! Have u recovery the result?

**Comment by jacedang** [comment]
@johnnylu305 May I ask a question, in your train code, where did you define the dataset dir (both train and val). I couldn't figure it out. Would appreciate if you can shed some light on it.

**Comment by johnnylu305** [comment]
@jacedang You can check [this](https://github.com/johnnylu305/CAR/commit/872a7611a26a12f88134c1d5695510c1d80bfc7d), I think it is in data/DIV2K_dataset.py  or data/__init__.py or others.

**Comment by jacedang** [comment]
@johnnylu305 I know it should be somewhere in DIV2K_dataset.py But I'm not quite sure what are LQ_root and GT_root. and where is the correct way to put data dir? Could you be more elaborate? Thank you.

**Comment by brighteast** [comment]
@johnnylu305 hi, really thank you for your code, but i have question. You just use L1 loss in train code. However, in a paper there are 3 loss including TV loss, offset constraint loss. So... i wonder if the results using your code are same with the paper.
